demos/cms/cms_ver.c

```diff
@@ -12,6 +12,53 @@
 #include <openssl/cms.h>
 #include <openssl/err.h>
 
+/*
+ * print any signingTime attributes.
+ * signingTime is when each party purportedly signed the message.
+ */
+static void print_signingTime(CMS_ContentInfo *cms)
+{
+    STACK_OF(CMS_SignerInfo) *sis;
+    CMS_SignerInfo *si;
+    X509_ATTRIBUTE *attr;
+    ASN1_TYPE *t;
+    ASN1_UTCTIME *utctime;
+    ASN1_GENERALIZEDTIME *gtime;
+    BIO *b;
+    int i, loc;
+
+    b = BIO_new_fp(stdout, BIO_NOCLOSE);
+    sis = CMS_get0_SignerInfos(cms);
+    for (i = 0; i < sk_CMS_SignerInfo_num(sis); i++) {
+        si = sk_CMS_SignerInfo_value(sis, i);
+        loc = CMS_signed_get_attr_by_NID(si, NID_pkcs9_signingTime, -1);
+        attr = CMS_signed_get_attr(si, loc);
+        if (X509_ATTRIBUTE_count(attr) > 1) {
```

Comment:
Dont know if the check is needed? The sign and verify both call ossl_cms_si_check_attributes() which check this

Reply:
I will drop it.

This code is meant only as an example of how to retrieve and display `signingTime` attributes.  There are a number of restrictions presented in RFC 5652 (section 11.3) that I am deliberately ignoring.